### PR TITLE
jamovi*: add r-pkgs depends

### PR DESCRIPTION
**Pull request by Kiri2002**
add r-pkgs depended by jamovi
using cran2pkgbuild.py

## Involved packages

 - Packages_Name

## Involved issue

Close #

## Details

- [ ] Tested in the local machine (largest 16G RAM) and it is passed without any issue
- [ ] Provide New Package
- [ ] Fix the Packages
  - [ ] PKGBUILD
  - [ ] lilac.yaml
  - [ ] lilac.py
- [ ] Would like to continue to work with us

### CODE REVIEW

File: BioArchLinux/r-rprotobuf/lilac.yaml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+build_prefix: extra-x86_64
+maintainers:
+- github: kiri2002
+repo_depends:
+- r-rcpp
+update_on:
+- regex: RProtoBuf_([\d._-]+).tar.gz
+  source: regex
+  url: https://cran.r-project.org/package=RProtoBuf
```
Comment (pekkarr) [inline BioArchLinux/r-rprotobuf/lilac.yaml:9]:
```suggestion
  url: https://cran.r-project.org/package=RProtoBuf
- source: alpm
  alpm: protobuf
  repo: extra
  provided: libprotobuf.so
  strip_release: true
```
Rebuild config for protobuf

**Comment by sukanka** [comment]
Do you plan to push these pkgs to aur?

**Review by pekkarr — CHANGES_REQUESTED** [review]
Please add `- alias: r` to the `update_on` config in all `lilac.yaml`s.

Also I'm not completely sure, but I think you need a `lilac.py` for updating the `_pkgver` variable. See [r-v8 lilac.py](../../blob/master/BioArchLinux/r-v8/lilac.py) for example.

The new metadata checking can be done later when it has been tested more.

File: BioArchLinux/r-rprotobuf/lilac.yaml (hunk 1 above)
Comment (starsareintherose) [reply BioArchLinux/r-rprotobuf/lilac.yaml:9]: Better to use cran instead of regex. @pekkarr now is lilac has the ability to handle the cran archive?

Comment (pekkarr) [reply BioArchLinux/r-rprotobuf/lilac.yaml:9]: Yes, the `cran` source would be better. However the new `rpkgs` source in archlinuxcn/lilac#201 is even more efficient than the `cran` source, but the pr hasn't been merged yet.

Comment (starsareintherose) [reply BioArchLinux/r-rprotobuf/lilac.yaml:9]: okay, I see

**Comment by starsareintherose** [comment]
Should these pkgs add lilac.py related content?

**Comment by pekkarr** [comment]
I think the [`lilac.py` in r-v8](../blob/master/BioArchLinux/r-v8/lilac.py) should be copied to every one of these packages. As far as I know lilac won't update the `_pkgver` variable otherwise. Also these packages should go to AUR as well, which requires the `update_aur_repo()` call.

**Comment by sukanka** [comment]
> I think the [`lilac.py` in r-v8](../blob/master/BioArchLinux/r-v8/lilac.py) should be copied to every one of these packages. As far as I know lilac won't update the `_pkgver` variable otherwise. Also these packages should go to AUR as well, which requires the `update_aur_repo()` call.


They can be moved to `lilac.yaml`. Anyway, we have to write these lines again and again.

```
pre_build_script: |
  for line in edit_file('PKGBUILD'):
    if line.startswith('_pkgver='):
      line = f'_pkgver={_G.newver}'
    print(line)
  update_pkgver_and_pkgrel(_G.newver.replace(':', '.').replace('-', '.'))
```

**Comment by pekkarr** [comment]
> They can be moved to `lilac.yaml`. Anyway, we have to write these lines again and again.

Yeah this works too (also the `post_bulid_script` should be added)

The new [`lilac_r_utils`](../blob/master/lilac-extensions/lilac_r_utils.py#L7) has the `r_update_pkgver_and_pkgrel` function that does just that, but it's better to not use it yet since I plan to change the API a bit.

**Comment by pekkarr** [comment]
The `rpkgs` nvchecker source was merged into lilac, so it should be used instead of regex (the lilac package has to be updated first).

The correct `update_on` config would look like this for `r-exact2x2` as an example:
```yaml
- source: rpkgs
  pkgname: exact2x2
  repo: cran
```
See also the [documentation](../../../archlinuxcn/lilac/tree/master/nvchecker_source#r-packages-from-cran-and-bioconductor) in lilac.

**Comment by starsareintherose** [comment]
After current upgrade, we need to upgrade lilac on our server to accept this change

**Comment by Kiri2002** [comment]
> Do you plan to push these pkgs to aur?

Nope

**Comment by Kiri2002** [comment]
all lilac.yaml chaned:
post_build_script: |
  git_pkgbuild_commit()
update_on:
- source: rpkgs
  pkgname: xxx
  repo: cran
- alias: r

**Comment by pekkarr** [comment]
Please change the other packages too, so far you've changed only `r-exact2x2`, `r-exactci` and `r-ssanv`. Also you'll need to add a `pre_build_script` like the one in [sukanka's comment](#issuecomment-1575498864).

Please also add the protobuf rebuild config for `r-rprotobuf`, after `- alias: r`.
```yaml
- source: alpm
  alpm: protobuf
  repo: extra
  provided: libprotobuf.so
  strip_release: true
```
This makes sure that the package is rebuilt when protobuf is updated to an incompatible version.

**Comment by Kiri2002** [comment]
> Please change the other packages too, so far you've changed only `r-exact2x2`, `r-exactci` and `r-ssanv`. Also you'll need to add a `pre_build_script` like the one in [sukanka's comment](#issuecomment-1575498864).
> 
> Please also add the protobuf rebuild config for `r-rprotobuf`, after `- alias: r`.
> 
> ```yaml
> - source: alpm
>   alpm: protobuf
>   repo: extra
>   provided: libprotobuf.so
>   strip_release: true
> ```
> 
> This makes sure that the package is rebuilt when protobuf is updated to an incompatible version.

done

File: BioArchLinux/r-pmcmr/lilac.yaml [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  for line in edit_file('PKGBUILD'):
+    if line.startswith('_pkgver='):
+      line = f'_pkgver={_G.newver}'
+    print(line)
+  update_pkgver_and_pkgrel(_G.newver.replace(':', '.').replace('-', '.'))
+post_build_script: |
+  git_pkgbuild_commit()
+update_on:
+- source: rpkgs
+  pkgname: pmcmr
```
Comment (pekkarr) [inline BioArchLinux/r-pmcmr/lilac.yaml:14]:
```suggestion
  pkgname: PMCMR
```
the name is case-sensitive

File: BioArchLinux/r-rprotobuf/lilac.yaml [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  for line in edit_file('PKGBUILD'):
+    if line.startswith('_pkgver='):
+      line = f'_pkgver={_G.newver}'
+    print(line)
+  update_pkgver_and_pkgrel(_G.newver.replace(':', '.').replace('-', '.'))
+post_build_script: |
+  git_pkgbuild_commit()
+update_on:
+- source: rpkgs
+  pkgname: rprotobuf
```
Comment (pekkarr) [inline BioArchLinux/r-rprotobuf/lilac.yaml:16]:
```suggestion
  pkgname: RProtoBuf
```

File: BioArchLinux/r-pmcmr/lilac.yaml (hunk 2 above)
Comment (starsareintherose) [reply BioArchLinux/r-pmcmr/lilac.yaml:14]: How patient you are!

**Comment by pekkarr** [comment]
I think this is ready for merge

**Pull request merged**